pkg/resourceinterpreter/defaultinterpreter/retain.go

```diff
@@ -111,5 +112,17 @@ func retainJobSelectorFields(desired, observed *unstructured.Unstructured) (*uns
 			return nil, err
 		}
 	}
+
+	// retain job completions: spec.completions is immutable if job was created before.
+	completions, ok, err := unstructured.NestedInt64(observed.Object, "spec", "completions")
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve completions from job: %v", err)
+	}
+	if ok {
+		if err = unstructured.SetNestedField(desired.Object, completions, "spec", "completions"); err != nil {
+			return nil, fmt.Errorf("failed to set completions for job: %v", err)
+		}
+	}
```

Comment:
Does this change involve problems that existed before?

Reply:
I think so, we had this problem before because job completion should never be updated.